src/db/models/device.rs

```diff
@@ -212,4 +212,16 @@ impl Device {
                 .from_db()
         }}
     }
+
+    pub async fn check_user_has_push_device(user_uuid: &str, conn: &mut DbConn) -> bool {
+        db_run! { conn: {
+            diesel::select(diesel::dsl::exists(
+                devices::table
+                    .filter(devices::user_uuid.eq(user_uuid))
+                    .filter(devices::push_token.is_not_null())
+            ))
```

Comment:
This format is not how we build the queries. There is some macro magic done which might break here in the future or other database connections.

Best would be to use the same format as the function above.

```suggestion
            devices::table
                .filter(devices::user_uuid.eq(user_uuid))
                .filter(devices::push_token.is_not_null())
```

Reply:
GitHub shows the suggestions in a broken way, but I think this is implemented now.